Automattic/wp-calypso
# Reimplement gridicons using SVG external content.

Reimplement gridicons using an SVG external content approach, instead of SVG transpiled into React components. This helps keep images as images (which is good for performance), while maintaining styleability. Old browsers are supported through the already-included svg4everyone polyfill.

This PR supersedes and replaces #32172.

See #32364 for a similar change, which added Material Icons with the same approach.

#### Changes proposed in this Pull Request

* Remove AsyncGridicons component.
* Add Gridicons component, using offset data and SVG sprite from `gridicons` NPM package.
* Bump gridicons NPM package version to one that includes the new files.
* Modify gridicon CSS rules to be compatible with this approach.

#### Testing instructions

* Open the calypso.live branch of this PR in one browser tab.
* Open production in another browser tab.
* Alternate between both tabs, comparing as many different Calypso pages as you see fit. They should all be identical, pixel for pixel.
